### PR TITLE
Use importlib.metadata instead of pkg_resources.

### LINKED ISSUE #452: Opening pyramidal images ob google collab: large_image getTileSource doesn't work on google collab (No available tilesource for ndpi)
I used all of these commands to make sure I have everything that's needed for loading dpi pyramidal images on google collab with no luck so far.

    !pip install large_image
    !apt update && apt install -y openslide-tools
    !pip install openslide-python
    !pip install libtiff openslide_python pyvips gdal mapnik pyproj glymur -f https://girder.github.io/large_image_wheels
!pip install large-image-source-openslide

I get the following error:

    ---------------------------------------------------------------------------
    TileSourceException                       Traceback (most recent call last)
    <ipython-input-23-0584be72095d> in <module>()
         27 print(os.listdir(ndpi_dir))
         28 print(target_filename)
    ---> 29 ts = large_image.getTileSource(target_filename)
         30 
         31 # print(ts.getMetadata())
    
    1 frames
    /usr/local/lib/python3.6/dist-packages/large_image/tilesource/__init__.py in getTileSourceFromDict(availableSources, pathOrUri, *args, **kwargs)
         65         if availableSources[sourceName].canRead(sourceObj, *args, **kwargs):
         66             return availableSources[sourceName](sourceObj, *args, **kwargs)
    ---> 67     raise TileSourceException('No available tilesource for %s' % pathOrUri)
         68 
         69 
    
    TileSourceException: No available c for ndpi_images/Li88TDCLAMP.ndpi

I made sure the the file is there, it's loading okay in other platforms using the same command, so it's not the file. 

How can I resolve this issue?

Thanks

### PR DESCRIPTION
**Pull request by manthey**
Resolves #452.

### CODE REVIEW

File: large_image/tilesource/__init__.py [hunk 1]
```diff
@@ -51,7 +49,12 @@ def loadTileSources(entryPointName='large_image.source', sourceDict=AvailableTil
     :param entryPointName: the name of the entry points to load.
     :param sourceDict: a dictionary to populate with the loaded sources.
     """
-    for entryPoint in iter_entry_points(entryPointName):
+    try:
+        from importlib.metadata import entry_points
+    except ImportError:
+        from importlib_metadata import entry_points
```
Comment (banesullivan) [inline large_image/tilesource/__init__.py:55]: If this is now a hard dependency and `importlib.metadata` is a part of stdlib >=3.8, then, in my opinion, this should be with the rest of the imports at the top of the file

File: setup.py [hunk 2]
```diff
@@ -77,6 +77,7 @@ def prerelease_local_scheme(version):
         'Pillow!=8.3.0,!=8.3.1',
         'psutil>=4.2.0',  # technically optional
         'numpy>=1.10.4',
+        'importlib-metadata ; python_version < "3.8"',
```
Comment (banesullivan) [inline setup.py:80]: @manthey, on the next release, I need to make sure this is captured in the conda recipe for py3.7

Comment (banesullivan) [reply setup.py:80]: Mostly a reminder for myself, and hoping you'll remind me if I forget

File: large_image/tilesource/__init__.py (hunk 1 above)
Comment (manthey) [reply large_image/tilesource/__init__.py:55]: Okay.  I'll shift it up.

**Pull request merged**